File: bindings/python/ailoy/models/tvm.py
from typing import Literal, Optional

from pydantic import BaseModel, Field

TVMModelId = Literal["Qwen/Qwen3-0.6B", "Qwen/Qwen3-1.7B", "Qwen/Qwen3-4B", "Qwen/Qwen3-8B"]

Quantization = Literal["q4f16_1"]


class TVMModel(BaseModel):
    component_type: str = Field("tvm_language_model", frozen=True, init=False)
    id: TVMModelId
    quantization: Quantization = "q4f16_1"
    device: int = 0

    @property
    def default_system_message(self) -> Optional[str]:
        if self.id.startswith("Qwen"):
            return "You are Qwen, created by Alibaba Cloud. You are a helpful assistant."
        return None

    def to_attrs(self):
        return {
            "model": self.id,
            "quantization": self.quantization,
            "device": self.device,
        }

File: bindings/js-node/src/models/tvm.ts
export type TVMModelId =
  | "Qwen/Qwen3-8B"
  | "Qwen/Qwen3-4B"
  | "Qwen/Qwen3-1.7B"
  | "Qwen/Qwen3-0.6B";

export type Quantization = "q4f16_1";

interface TVMModelArgs {
  id: TVMModelId;
  quantization?: Quantization;
  device?: number;
}

export class TVMModel {
  id: TVMModelId;
  quantization: Quantization;
  device: number;
  readonly componentType: string = "tvm_language_model";

  constructor(args: TVMModelArgs) {
    this.id = args.id;
    this.quantization = args.quantization ?? "q4f16_1";
    this.device = args.device ?? 0;
  }

  defaultSystemMessage(): string | undefined {
    if (this.id.startsWith("Qwen")) {
      return "You are Qwen, created by Alibaba Cloud. You are a helpful assistant.";
    }
    return undefined;
  }

  toAttrs() {
    return {
      model: this.id,
      quantization: this.quantization,
      device: this.device,
    };
  }
}

export default function (args: TVMModelArgs): TVMModel {
  return new TVMModel(args);
}
